### PR TITLE
Matrix convert o zero - initial checkin

### PR DESCRIPTION
**Pull request by shrutivanw**
Exercise 6 from the older [integers and arrays CS Fundamentals assignment](https://github.com/AdaGold/IntegersAndArrays).
Leverages mini tests for consistency with other CS Fundamentals' homework.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,20 @@
-# matrix_convert_to_zero
\ No newline at end of file
+# Matrix: Convert to zero
+Changes the corresponding row and column to have '0' values if the value is '0' in the input matrix.
+
+## Exercise
+* Design and implement a method that updates the input matrix. The input matrix i.e. two-dimensional array contains only '0's and '1's. If any number in the array is found to be '0', the method updates all the numbers in the corresponding row as well as the corresponding column to be '0'.
+* For example if the input is:
```
Comment (Hamled) [inline README.md:6]: I think you'll need to use a code block (with three back ticks), or possibly a table, to display this example properly. Currently the whole thing is rendered onto a single line. For example using code blocks:

The input matrix
```
1 1 1
1 0 1
1 1 1
1 1 0
```
becomes
```
1 0 0
0 0 0
1 0 0
0 0 0
```

File: README.md [hunk 2]
```diff
@@ -1 +1,20 @@
...
+should get updated to:
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1 0 0
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0 0 0
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1 0 0
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0 0 0
+
+* Share and explain the time and space complexities for your solution.
+* If you think of multiple approaches to solve the problem, implement the solution which minimizes space complexity. Explain the other approaches, and your decision in comments above the code.
+
+**Note**: Do not use Ruby provided functionality.
```
Comment (Hamled) [inline README.md]: What functionality are we talking about here?

I could definitely see this algorithm being something that they write using standard Ruby constructs such as `each` or other enumerable methods, and I'm not sure that would detract from their understanding.

I don't know if there are any Ruby libraries for dealing with matrices (I assume there are some Gems, not sure about in the standard lib), but that would be the most obvious thing to restrict them from using.

**Review by Hamled — CHANGES_REQUESTED** [review]

**Comment by Hamled** [comment]
Another thought I have is... since at least some of this exercise is to get them thinking about complexity... shouldn't we have some tests that are actually large in scale?

If we write a method that takes an array of `[row, column]` coordinates for the zeros, we should be able to programmatically determine the rows and columns which need to be fully zero'd after the method is called. We could then generate input matrices of arbitrary size (10,000 x 10,000 or larger), generate random coordinates within there to zero, and automatically verify the result.

This would, I think, give a more tangible experience to how different algorithms perform.

File: README.md (hunk 1 above)
Comment (shrutivanw) [reply README.md:6]: Done.

File: README.md (hunk 2 above)
Comment (shrutivanw) [reply README.md]: Fair point! Removed.

**Comment by shrutivanw** [comment]
Good suggestion, Charles! I'm currently struggling with adding that test case in a way that doesn't give away one possible solution to the problem itself. So, for now, I'm tracking a new issue, #2 to add a new test case with a large sized matrix in the future. Just buying more time to get that test case added more elegantly and unblock C9 for now.

Addressed the other feedback. Thank you!

**Review by Hamled — APPROVED** [review]

**Pull request merged**